src/datasets/loaders/scrnaseq/op3_loader/script.py

```diff
@@ -0,0 +1,302 @@
+    logger.info("Adding metadata to .uns")
+    adata.uns['dataset_id'] = par["dataset_id"]
+    adata.uns['dataset_name'] = par["dataset_name"]
+    adata.uns['dataset_summary'] = par["dataset_summary"]
+    adata.uns['dataset_description'] = par["dataset_description"]
+    adata.uns['dataset_organism'] = 'Homo sapiens'
+    adata.uns['dataset_url'] = 'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE279945'
+    adata.uns['dataset_reference'] = 'GSE279945'
+    adata.uns['dataset_version'] = '1.0.0'
+    adata.uns['processing_status'] = 'processed'
+
+def print_unique(adata, column):
+    """Print unique values in a column."""
+    if column in adata.obs:
+        values = adata.obs[column].unique()
+        if len(values) <= 10:
+            formatted = "', '".join(values)
+            logger.info(f"Unique {column}: ['{formatted}']")
+        else:
+            logger.info(f"Unique {column}: {len(values)} values")
+
+def print_summary(adata):
+    """Print a summary of the dataset."""
+    logger.info(f"Dataset shape: {adata.shape}")
+    logger.info(f"Number of cells: {adata.n_obs}")
+    logger.info(f"Number of genes: {adata.n_vars}")
+    
+    # Print unique values for key columns
+    for column in ['donor_id', 'cell_type', 'perturbation']:
+        print_unique(adata, column)
+        
+    logger.info(f"Layers: {list(adata.layers.keys())}")
+    logger.info(f"Metadata: {list(adata.uns.keys())}")
+
+def write_anndata(adata, par):
+    """Write AnnData object to file."""
+    logger.info(f"Writing AnnData object to '{par['output']}'")
+    adata.write_h5ad(par["output"], compression=par["output_compression"])
+
+def main(par, meta):
+    """Main function."""
+    logger.info("Starting OP3 loader")
+    
+    # Download the data
+    data_path = download_op3_data()
+    
+    # Load the data
+    logger.info(f"Loading data from {data_path}")
+    adata = sc.read_h5ad(data_path)
+    
+    # Apply OP3-specific filtering
+    adata = filter_op3_data(adata)
+    
+    # Filter by parameters
+    if par["donor_id"] is not None:
+        logger.info(f"Filtering for donor_id: {par['donor_id']}")
+        adata = adata[adata.obs["donor_id"] == par["donor_id"]]
+    
+    if par["cell_type"] is not None:
+        logger.info(f"Filtering for cell_type: {par['cell_type']}")
+        adata = adata[adata.obs["cell_type"] == par["cell_type"]]
+    
+    if par["perturbation"] is not None:
+        logger.info(f"Filtering for perturbation: {par['perturbation']}")
+        adata = adata[adata.obs["perturbation"] == par["perturbation"]]
+    
+    # Filter cells and genes
+    adata = filter_by_counts(adata, par)
+    
+    # Move X to layers and normalize
+    move_x_to_layers(adata)
+    
+    # Add dataset metadata
+    add_metadata_to_uns(adata, par)
+    
+    # Print summary and save
+    print_summary(adata)
+    write_anndata(adata, par)
+    
+    logger.info("Done")
+
+if __name__ == "__main__":
```

Comment:
no need to define main as a separate function and calling it here - i.e. the main code should be written directly in scripts

Reply:
can stay the way it is